src/core/socialprovider.js

```diff
@@ -45,7 +45,7 @@ var XMPPSocialProvider = function(dispatchEvent) {
   // rate limiting).
   this.sendMessagesTimeout = null;
   this.timeOfFirstMessageInBatch = 0;
-  this.messages = [];
+  this.messages = {};
```

Comment:
I'd add a comment to let the reader know what the type/structure is of messages. Namely that `this.messages` is indexed by `clientId`'s, and each value is an array of strings that are the (XML-escaped? or is that done in `client.send`?) messages to sent to that client-id.